### PR TITLE
mocktikv: add write op_lock for mocktikv

### PR DESCRIPTION
**Pull request by cfzjywxk**
### What problem does this PR solve? 

Do commit `Op_Lock` type in `mocktikv`,  make it compatible with `tikv`

### Check List 

Tests 

 - Unit test

Code changes

Side effects

Related changes

Release note

### CODE REVIEW

File: server/http_handler_test.go [hunk 1]
```diff
@@ -440,7 +440,7 @@ func (ts *HTTPHandlerTestSuite) TestGetTableMVCC(c *C) {
 	info := data.Value.Info
 	c.Assert(info, NotNil)
 	c.Assert(len(info.Writes), Greater, 0)
-	startTs := info.Writes[0].StartTs
+	startTs := info.Writes[2].StartTs
```
Comment (jackysp) [inline server/http_handler_test.go]: bug fix?

File: session/session_test.go [hunk 2]
```diff
@@ -1996,7 +1996,7 @@ func (s *testSchemaSerialSuite) TestSchemaCheckerSQL(c *C) {
 	tk1.MustExec(`alter table t add index idx3(c);`)
 	tk.MustQuery(`select * from t for update`)
 	_, err = tk.Exec(`commit;`)
-	c.Assert(terror.ErrorEqual(err, domain.ErrInfoSchemaChanged), IsTrue, Commentf("err %v", err))
+	c.Assert(err, NotNil)
```
Comment (jackysp) [inline session/session_test.go]: The error is write conflict now?

File: server/http_handler_test.go (hunk 1 above)
Comment (cfzjywxk) [reply server/http_handler_test.go]: There will be 4 records returned(the nearest 2 are `op_lock` records), using the `op_lock` record `startTS` may fetch the index key, so using the row key as before

File: session/session_test.go (hunk 2 above)
Comment (cfzjywxk) [reply session/session_test.go]: Yes

Comment (tiancaiamao) [reply session/session_test.go]: `select * from t where id = 666 for update`
Could this one avoid the conflict error?

Comment (cfzjywxk) [reply session/session_test.go]: Yes, using this will get `schema has changed` error

**Comment by cfzjywxk** [comment]
@coocood @lysu PTAL

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**